src/conventions-and-definitions.md

```diff
@@ -14,9 +14,9 @@ The `Curve25519` function, which multiplies a curve (or twist) point by a "scala
 secret key), is defined in the paper. The `clamp_scalar` function is defined as follows:
 
     fn clamp_scalar(scalar: [u8; 32]) -> [u8; 32] {
-        scalar[0] &= 248
-        scalar[31] &= 127
-        scalar[31] |= 64
+        scalar[0] &= 248 // TODO paper show {0, 8, 16, 24, . . . , 248}, I understand 7 different values, something like 1 << (salar[0] % 7) 
+        scalar[31] &= 127 // same as % 127
+        scalar[31] |= 64 // TODO: {64, 65, 66, . . . , 127} so we got {0...127}, here we do | 64 making simply +64 so we actually want initial range to b {0..63, since 63 is 0b11111 we should scalar[31] &= 63 instead of &= 127
```

Comment:
This line just forces the second-highest bit on. The only difference between &= 63 and &= 127 is what happens to the second-highest bit, so either would work.

FWIW this is the standard X25519 clamping that you can see in eg [`curve25519-dalek`](https://github.com/dalek-cryptography/curve25519-dalek/blob/e44d4b5903106dde0e5b28a2580061de7dfe8a9f/curve25519-dalek/src/scalar.rs#L1243).

Reply:
well I don't get this one. from line 18 we got 0..127 (clearing to bit indeed), then we | 64, so if the value was 0 it become 64, but if the value was 1..64 it become also 64, sounds like the random distribution is wrong.
if previous line was &= 63 then we would have 0..63 range and then just need to do +64.